### PR TITLE
Change CUDAWrappers::MatrixFree::point_type to Point

### LINKED ISSUE #8240: CUDAWrappers::MatrixFree::point_type
The `CUDAWrappers::MatrixFree` class has this local typedef:
```
  template <int dim, typename Number = double>
  class MatrixFree : public Subscriptor
  {
  public:
    using jacobian_type = Tensor<2, dim, Tensor<1, dim, Number>>;
    // TODO this should really be a CUDAWrappers::Point
    using point_type = Tensor<1, dim, Number>;
```
I had to look this up when I saw it in step-64. The question I have is: Why is this not `Point<dim,Number>`? That's what we usually use to denote points in `dim` dimensional space, such as quadrature points.

In particular, the current definition leads to this function returning an apparently wrong data type:
```
  template <int dim, typename Number>
  __device__ inline typename CUDAWrappers::MatrixFree<dim, Number>::point_type &
  get_quadrature_point(
    const unsigned int                                          cell,
    const typename CUDAWrappers::MatrixFree<dim, Number>::Data *data,
    const unsigned int                                          n_q_points_1d)
  {
    return *(data->q_points + data->padding_length * cell +
             q_point_id_in_cell<dim>(n_q_points_1d));
  }
```
In step-64, this is then used as follows:
```
  const Tensor<1, dim> q_point =
    CUDAWrappers::get_quadrature_point<dim, double>(cell,
                                                    gpu_data,
                                                    n_dofs_1d);
```

### PR DESCRIPTION
**Pull request by masterleinad**
Fixes #8240. This compiles and passes all `cuda` tests for me using `CUDA-8`.

### CODE REVIEW

File: include/deal.II/matrix_free/cuda_matrix_free.h [hunk 1]
```diff
@@ -78,8 +78,7 @@ namespace CUDAWrappers
   {
   public:
     using jacobian_type = Tensor<2, dim, Tensor<1, dim, Number>>;
-    // TODO this should really be a CUDAWrappers::Point
-    using point_type = Tensor<1, dim, Number>;
+    using point_type    = Point<dim, Number>;
```
Comment (Rombur) [inline include/deal.II/matrix_free/cuda_matrix_free.h]: Can you grep for another `FIXME` or `TODO` about `CUDAWrappers::Point`, I think that there is another one about it.

Comment (masterleinad) [reply include/deal.II/matrix_free/cuda_matrix_free.h]: I can't find anything.

Comment (Rombur) [reply include/deal.II/matrix_free/cuda_matrix_free.h]: Alright thanks

**Pull request merged**